hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/producer/kafka/KafkaProducer.java

```diff
@@ -0,0 +1,166 @@
+package pl.allegro.tech.hermes.frontend.producer.kafka;
+
+import org.apache.kafka.clients.consumer.ConsumerGroupMetadata;
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.errors.ProducerFencedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.frontend.metric.CachedTopic;
+import pl.allegro.tech.hermes.frontend.producer.BrokerLatencyReporter;
+import pl.allegro.tech.hermes.frontend.publishing.message.Message;
+import pl.allegro.tech.hermes.frontend.publishing.metadata.ProduceMetadata;
+import pl.allegro.tech.hermes.metrics.HermesTimerContext;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+
+// TODO: is it a good idea to create abstraction on top of Producer? Currently it provides DC and instrumentation
```

Comment:
I'm fine with wrapping an instance of `org.apache.kafka.clients.producer.Producer` in something that adds instrumentation and datacenter awareness, but implementing the entire interface seems overkill. Maybe, let's just add a class that exposes only necessary methods (`send`, `close`, `partitionsFor`) + adds the mentioned features in the comment. Additionally, I'd refrain from calling the wrapper KafkaProducer. In my opinion, this name is "reserved" for the Kafka producer provided by the Kafka client library. I guess something like `KafkaMessageSender` is good enough.

Reply:
> I'm fine with wrapping an instance of org.apache.kafka.clients.producer.Producer in something that adds instrumentation and datacenter awareness, but implementing the entire interface seems overkill. Maybe, let's just add a class that exposes only necessary methods (send, close, partitionsFor) + adds the mentioned features in the comment

You are right, there is no point in delegating all the methods when only some of them are used. It also very confusing that there would be 3 available send methods. Left only those which are needed.

I've also change the name to `KafkaMessageSender` but this resulted in `Producers` being changed to `KafkaMessageSenders`  for consistency